pkgs/development/libraries/glibc/common.nix

```diff
@@ -133,6 +134,10 @@ stdenv.mkDerivation ({
       # nscd needs libgcc, and we don't want it dynamically linked
       # because we don't want it to depend on bootstrap-tools libs.
       echo "LDFLAGS-nscd += -static-libgcc" >> nscd/Makefile
+
+      # https://github.com/NixOS/nixpkgs/issues/124401
+      substituteInPlace resolv/netdb.h \
+            --replace /etc/protocols ${netbase}/etc/protocols
```

Comment:
Debugged the build and here was the output:

```
#define _PATH_NETWORKS          "/etc/networks"
#define _PATH_NSSWITCH_CONF     "/etc/nsswitch.conf"
#define _PATH_PROTOCOLS         "/nix/store/5bx3ljs16wvm94v04mrl1ydpgy4y8500-netbase-v6.3/etc/protocols"
#define _PATH_SERVICES          "/etc/services"
```

Reply:
Is it possible to fallback to this file if `/etc/protocols` is missing?

Reply:
this #define isn't even enough -- I need to revisit this change and find out where the _read_ happens and do the check there.
Your fallback would be the best case scenario

Reply:
Removed this line.